### PR TITLE
Add fixup for copying configuration

### PR DESCRIPTION
**Pull request by ocallesp**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/project-system/pull/6300)

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/CloneConfigurationFixupFactory.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using Microsoft.VisualStudio.ProjectSystem.VS.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    /// <summary>
+    /// Interface for components that provide post-configuration cloning property fix ups.
+    /// </summary>
+    [Export(typeof(IClonePlatformFixupFactory))]
+    [AppliesTo(ProjectCapabilities.Managed)]
+    internal class CloneConfigurationFixupFactory : IClonePlatformFixupFactory
```
Comment (ocallesp) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/CloneConfigurationFixupFactory.cs:13]: It seems I need to create another interface ICloneConfigurationFixupFactory and ICloneConfigurationFixup.

In CPS, how to differentiate platform and configuration fixups ?
@davkean

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/CloneConfigurationFixupFactory.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.ProjectSystem.VS.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    /// <summary>
+    /// Interface for components that provide post-configuration cloning property fix ups.
+    /// </summary>
+    [Export(typeof(IClonePlatformFixupFactory))]
+    [AppliesTo(ProjectCapabilities.Managed)]
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/CloneConfigurationFixupFactory.cs:12]: This needs to be "DotNet", Managed includes C++ for C++/CLI support.

Comment (ocallesp) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/CloneConfigurationFixupFactory.cs:12]: Done in next commit. 

Changed to ProjectCapability.DotNet

File: build/import/Packages.targets [hunk 3]
```diff
@@ -4,6 +4,7 @@
   <PropertyGroup>
     <RestoreSources>
       $(RestoreSources);
+      D:\PROJECT\cps\bin\Debug\packages\nuget;
```
Comment (ocallesp) [inline build/import/Packages.targets:7]: ignore this

File: build/import/Packages.targets [hunk 4]
```diff
@@ -95,8 +96,8 @@
     <PackageReference Include="VSSDK.TemplateWizardInterface"                                         Version="12.0.4"                          ExcludeAssets="All" />
 
     <!-- CPS -->
-    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.SDK"                               Version="16.7.232-pre" />
-    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.Analyzers"                         Version="16.7.232-pre" />
+    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.SDK"                               Version="16.7.325-pre-g177fdc035e" />
+    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.Analyzers"                         Version="16.7.325-pre-g177fdc035e" />
```
Comment (ocallesp) [inline build/import/Packages.targets:100]: ignore

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/CloneConfigurationFixup.cs [hunk 5]
```diff
@@ -11,76 +16,117 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
...
+            try
+            {
+                var ProjectLockService = _unconfiguredProjectServices.ActiveConfiguredProjectProvider.ActiveConfiguredProject
+                    .Services.ProjectLockService;
+
+                ProjectLockService.ReadLockAsync(
+                    async access =>
+                    {
+                        configuredProject =
+                            await _unconfiguredProject.LoadConfiguredProjectAsync(_fromProjectConfiguration);
```
Comment (ocallesp) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/CloneConfigurationFixup.cs:89]: LoadConfiguredProjectAsync returns null.

Am I using these API correctly ?
@davkean

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.CSharp.DesignTime.targets [hunk 6]
```diff
@@ -21,7 +21,7 @@
     <DesignerFunctionVisibility Condition="'$(DesignerFunctionVisibility)' == ''">Private</DesignerFunctionVisibility>
     <DesignerVariableNaming Condition="'$(DesignerVariableNaming)' == ''">Camel</DesignerVariableNaming>
     <DesignerHiddenCodeGeneration Condition="'$(DesignerHiddenCodeGeneration)' == ''">Declarations</DesignerHiddenCodeGeneration>
-  
+    <CopyingConfigurationProperties Condition="'$(CopyingConfigurationProperties)' == ''">Optimize;DebugSymbols</CopyingConfigurationProperties>
```
Comment (ocallesp) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.CSharp.DesignTime.targets:24]: I think this is not needed

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ConfigurationGeneral.xaml [hunk 7]
```diff
@@ -55,6 +55,9 @@
   <StringProperty Name="Configuration"
                   Visible="false" />
 
+  <StringProperty Name="CopyingConfigurationProperties"
```
Comment (ocallesp) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ConfigurationGeneral.xaml:58]: not needed

File: build/import/Packages.targets (hunk 3 above)
Comment (drewnoakes) [reply build/import/Packages.targets:7]: FYI you can do this locally as well via `NuGet.config`. In my `%appdata%\NuGet\NuGet.Config` I have something like:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="NuGet Local" value="D:\NuGetLocal" />
```

**Comment by ocallesp** [comment]
I will submit in another pr

**Pull request closed without merging**